cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/distributed/remote/RemoteExecutionTwillRunnerService.java

```diff
@@ -205,8 +207,13 @@ final void doInitialize() {
       throw new RuntimeException(e);
     }
 
-    scheduler = Executors.newScheduledThreadPool(cConf.getInt(Constants.RuntimeMonitor.THREADS),
-                                                 Threads.createDaemonThreadFactory("runtime-scheduler-%d"));
+    int schedulerThreads = cConf.getInt(Constants.RuntimeMonitor.THREADS);
+    ThreadFactory schedulerThreadFactory = Threads.createDaemonThreadFactory("runtime-scheduler-%d");
+    if (cConf.getBoolean(Constants.Security.EXPLICIT_CONTEXT_PROPAGATION_ENABLED)) {
```

Comment:
Are we going to set this to false if system workers are disabled, or is it safe to run through delegate in appfabric also?

Reply:
It should be safe in appfabric as well (in fact it would be more incorrect to not have this in appfabric).